src/go/rpk/pkg/cli/cmd/acl/common.go

```diff
@@ -57,26 +58,90 @@ type (
-		ResourceType        kmsg.ACLResourceType
-		ResourceName        string
-		ResourcePatternType kmsg.ACLResourcePatternType
-		Operation           kmsg.ACLOperation
-		Permission          kmsg.ACLPermissionType
+		Principal           string                      `json:"principal" yaml:"principal"`
+		Host                string                      `json:"host" yaml:"host"`
+		ResourceType        kmsg.ACLResourceType        `json:"resource_type" yaml:"resource_type"`
+		ResourceName        string                      `json:"resource_name" yaml:"resource_name"`
+		ResourcePatternType kmsg.ACLResourcePatternType `json:"resource_pattern_type" yaml:"resource_pattern_type"`
+		Operation           kmsg.ACLOperation           `json:"operation" yaml:"operation"`
+		Permission          kmsg.ACLPermissionType      `json:"permission" yaml:"permission"`
 	}
+
 	aclWithMessage struct {
-		Principal           string
-		Host                string
-		ResourceType        kmsg.ACLResourceType
-		ResourceName        string
-		ResourcePatternType kmsg.ACLResourcePatternType
-		Operation           kmsg.ACLOperation
-		Permission          kmsg.ACLPermissionType
-		Message             string
+		Principal           string                      `json:"principal" yaml:"principal"`
+		Host                string                      `json:"host" yaml:"host"`
+		ResourceType        kmsg.ACLResourceType        `json:"resource_type" yaml:"resource_type"`
+		ResourceName        string                      `json:"resource_name" yaml:"resource_name"`
+		ResourcePatternType kmsg.ACLResourcePatternType `json:"resource_pattern_type" yaml:"resource_pattern_type"`
+		Operation           kmsg.ACLOperation           `json:"operation" yaml:"operation"`
+		Permission          kmsg.ACLPermissionType      `json:"permission" yaml:"permission"`
+		Message             string                      `json:"message" yaml:"message"`
+	}
+
+	// Used for json/yaml formatted output in "rpk acl lists|delete".
+	describedACLsResult struct {
+		Principal *string `json:"principal" yaml:"principal"` // Principal is the optional user that was used in this filter.
+		Host      *string `json:"host" yaml:"host"`           // Host is the optional host that was used in this filter.
+
+		Type       kmsg.ACLResourceType   `json:"type" yaml:"type"`             // Type is the type of resource used for this filter.
+		Name       *string                `json:"name" yaml:"name"`             // Name is the name of the resource used for this filter.
+		Pattern    kadm.ACLPattern        `json:"pattern" yaml:"pattern"`       // Pattern is the name pattern used for this filter.
+		Operation  kmsg.ACLOperation      `json:"operation" yaml:"operation"`   // Operation is the operation used for this filter.
+		Permission kmsg.ACLPermissionType `json:"permission" yaml:"permission"` // Permission is permission used for this filter.
+
+		Err error `json:"error" yaml:"error"` // Err is non-nil if this filter has an error.
+	}
```

Comment:
`describedACLsResult` can be dropped in favor of `aclWithMessage`, which is the same thing